add-new-exercise.ps1

```diff
@@ -81,35 +87,67 @@ function Add-Project {
     $proj.Save($csProj);
 }
 
-function Add-Generator {
+function Add-GeneratorClass {
+    param (
+        [Parameter(Position = 0, Mandatory = $true)][string]$ExerciseName
+    )
+
     Write-Output "Adding generator"
 
     $generatorsDir = Resolve-Path "generators"
     $generatorsExercisesDir = Join-Path $generatorsDir "Exercises"
     $generatorsExerciseGeneratorsDir = Join-Path $generatorsExercisesDir "Generators"
-    $generator = Join-Path $generatorsExerciseGeneratorsDir "$exerciseName.cs"
-    $generatorClass = "namespace Exercism.CSharp.Exercises.Generators`n{`n    public class $exerciseName : GeneratorExercise`n    {`n    }`n}"
+    $generator = Join-Path $generatorsExerciseGeneratorsDir "$ExerciseName.cs"
+    $generatorClass = "namespace Exercism.CSharp.Exercises.Generators`n{`n    public class $ExerciseName : GeneratorExercise`n    {`n    }`n}"
     
     Set-Content -Path $generator -Value $generatorClass
 }
 
-function Copy-Track-Files {
-    Write-Output "Copying track files"
-    ./copy-track-files.ps1 $Exercise
+function Copy-TrackFile {
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Position = 0, Mandatory = $true)][string]$Exercise
+    )
+
+    if ($PSCmdlet.ShouldProcess("exercise $Exercise", "copy generic track files")) {
+        Write-Output "Copying track files"
+        ./copy-track-files.ps1 $Exercise
+    }
 }
 
 function Update-Readme {
-    Write-Output "Updating README"
-    ./update-docs.ps1 $Exercise
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Position = 0, Mandatory = $true)][string]$Exercise
+    )
+
+    if ($PSCmdlet.ShouldProcess("exercise $Exercise", "update Readme files")) {
+        Write-Output "Updating README"
+        ./update-docs.ps1 $Exercise
+    }
 }
 
-function Update-Tests { 
-    Write-Output "Updating test suite"
-    ./generate-tests.ps1 $Exercise
+function Update-Test {
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Position = 0, Mandatory = $true)][string]$Exercise
+    )
+
+    if ($PSCmdlet.ShouldProcess("exercise $Exercise", "generate new tests")) {
+        Write-Output "Updating test suite"
+        ./generate-tests.ps1 $Exercise
+    } 
 }
 
-function Update-Config-Json {
-    Write-Output "Updating config.json"
+function Update-ConfigJson {
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory = $true, Position = 0)][string]$Exercise,
+        [Parameter()][string[]]$Topics = @(),
+        [Parameter()][switch]$Core,
+        [Parameter()][int]$Difficulty = 1,
+        [Parameter()]$UnlockedBy
+    )
 
     $configJson = Resolve-Path "config.json"
 
```

Comment:
the config file changes, so currently this does not work

Reply:
Yeah, we need to do some tweaking later. Or possibly two separate scripts.